### PR TITLE
Calculator Unit Test

### PR DESCRIPTION
**Pull request by Matias-Villalobo**
image

All the Unit Tests for the calculator have been done.

### CODE REVIEW

**Review by mauroLonghi — COMMENTED** [review]
Mati, very good test, but Could you add a operation test with negative number please?

File: app/src/test/java/com/example/mycalculator/mvp/presenter/CalculatorPresenterTest.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        assertEquals(NUMBER_THREE, model.getPartialResult());
+    }
+
+    @Test
+    public void testNumberPressedSecondOperand() {
+        model.saveNumber(NUMBER_THREE);
+        model.saveOperationSymbol(OPERATOR_SUM);
+        presenter.numberPressed(NUMBER_THREE);
+        verify(view).drawNumber(model.getPartialResult());
+        assertEquals(NUMBER_THREE + OPERATOR_SUM + NUMBER_THREE, model.getPartialResult());
```
Comment (SartiMau) [inline app/src/test/java/com/example/mycalculator/mvp/presenter/CalculatorPresenterTest.java:108]: add an assert for the operation result besides the getPartialResult

File: app/src/test/java/com/example/mycalculator/mvp/presenter/CalculatorPresenterTest.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import static com.example.utils.StringUtilsTest.NUMBER_ZERO;
+import static com.example.utils.StringUtilsTest.OPERATOR_DIVIDE;
+import static com.example.utils.StringUtilsTest.OPERATOR_MINUS;
+import static com.example.utils.StringUtilsTest.OPERATOR_MULTIPLY;
+import static com.example.utils.StringUtilsTest.OPERATOR_SUM;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+public class CalculatorPresenterTest {
```
Comment (SartiMau) [inline app/src/test/java/com/example/mycalculator/mvp/presenter/CalculatorPresenterTest.java:25]: add a complete operation test for divide, subtract and multiply

**Review by SartiMau — CHANGES_REQUESTED** [review]

**Comment by Matias-Villalobo** [comment]
image

**Review by mauroLonghi — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Comment by Matias-Villalobo** [comment]
image

Now the names of the tests are clearer.

**Review by BQuinteros — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]
